### DISCUSSION TITLE
Issue with wxwidgets

### DISCUSSION THREAD

**Discussion by Kleb0** [xmake-io/xmake-repo#1390] [General]
hello, i hope you're doin good. First of all, i would like to thank you very much for your work on xmake, it's an incredible project !

But i have an issue to report : 

When i try to install wxwidgets, xmake seams to be unable to install the package. I test over packages with the follwing method in my xmake.lua file, and it work perfectly. 

By exemple when i put in my code : 
" add_requires("zlib-ng 2.0.6") 
  add_packages("zlib-ng") "

it works perfectly, xmake in cmd asked me if i want to install the package in the project. 

But, when i write : 
" add_requires("wxwidgets 3.2.0")
add_packages("wxwidgets") "
it doesn't install the package, and i can't add the include "wx/wx.h" to my projet.

here is my LUA file : 

https://github.com/ClementMiermont/Game_Engine_Project/blob/main/RPG_From_Scratch%20project%202/xmake.lua

Am i doing something wrong ?

**Comment by waruqi** [comment]
try add_requires("wxwidgets", {system = false})

and let me see the verbose logs

or look at https://github.com/xmake-examples/wxWidgets-scaffold

**Reply by Kleb0** [reply]
I tried this solution and i looked at the base template too, but it didn't work, in fact, nothing appears in the windows console

**Reply by waruqi** [reply]
```console
xmake f -cvD
xmake -rvD
```

let me see the output

**Reply by waruqi** [reply]
? I need full output.

**Reply by waruqi** [reply]
> ```
> xmake f -cvD
> xmake -rvD
> ```
> 
> let me see the output

this output

**Reply by Kleb0** [reply]
Well this is the output i got when i enter the commands. What do you mean by output ? You talk about sorted files in xmake directory or the output in Windows console ? Can you be more precise ?

**Reply by waruqi** [reply]
The output of these two commands in the windows console.

**Reply by Kleb0** [reply]
okay so it output this when i enter xmake f -cvD

![image](https://user-images.githubusercontent.com/91457036/183252819-7732f924-6702-4be8-8281-705ffc203778.png)

**Reply by waruqi** [reply]
I need full logs

**Reply by Kleb0** [reply]
![image](https://user-images.githubusercontent.com/91457036/183269078-caa1dc56-d830-4855-ba4f-74600ffd5a48.png)

**Reply by Kleb0** [reply]
![image](https://user-images.githubusercontent.com/91457036/183269097-13152ead-bb98-460d-a5ce-10f793d309e5.png)

**Reply by waruqi** [reply]
full logs of two commands, not only one.

pleae give me txt

**Reply by waruqi** [reply]
redirect output

**Reply by Kleb0** [reply]
[package CVD.txt](https://github.com/xmake-io/xmake-repo/files/9275919/package.CVD.txt)
[package rvD.txt](https://github.com/xmake-io/xmake-repo/files/9275922/package.rvD.txt)


here are the package txt files i got with xmake rvD and cvD command, it corresponds to the output logs ?

**Reply by Kleb0** [reply]
[detect.txt](https://github.com/xmake-io/xmake-repo/files/9275928/detect.txt)

Here is the detect txt file too

**Reply by waruqi** [reply]
> [package CVD.txt](https://github.com/xmake-io/xmake-repo/files/9275919/package.CVD.txt) [package rvD.txt](https://github.com/xmake-io/xmake-repo/files/9275922/package.rvD.txt)
> 
> here are the package txt files i got with xmake rvD and cvD command, it corresponds to the output logs ?

I saw they are both the content of all detect.txt

**Reply by Kleb0** [reply]
it's pretty weird, i can see that wxwidgets is set to false, but i follow the rules for installing package to the solution/project

**Reply by waruqi** [reply]
I need logs, but so far I only have the contents of a detect.txt.

I can't analyze the problem further, so you'll have to troubleshoot the cause yourself.

**Reply by Kleb0** [reply]
Man, i would like to give you the outputs logs, but the program is closing litteraly too fast, i don't even know how to give you txt file from the log in console. 

I follow exactly the same procedure for others packages in the repo, it works perfectly, but when i do it for wxwidgets, well it doesn't

**Reply by waruqi** [reply]
redirect output, `xmake f -vD &> xxx.log`

**Reply by waruqi** [reply]
If I don't have logs I can't help you much.

**Reply by waruqi** [reply]
xmake f -vD > xxx.log
xmake f -vD 2> xxx.log
...

you can google redirect output for windows

**Reply by Kleb0** [reply]
I got effectly an output file, but nothing is written on it

**Reply by Kleb0** [reply]
![image](https://user-images.githubusercontent.com/91457036/183270155-475f9c1b-3ec2-45b7-817d-39225929fef4.png)
![image](https://user-images.githubusercontent.com/91457036/183270161-d75247a0-cb31-43de-af8e-54afe72a30dc.png)

**Reply by waruqi** [reply]
xmake f -cvD -y

**Reply by Kleb0** [reply]
well, still nothing in the output.log file

**Reply by waruqi** [reply]
xmake f -cvD -y > xxx.log

**Reply by Kleb0** [reply]
still nothing in the output txt file

**Reply by Kleb0** [reply]
sorry to bother you, it's probably me who is doing something wrong, but i really don't know why this don't work, i follow exactly the procedure for the lua file.

**Reply by Kleb0** [reply]
> xmake f -cvD -y > xxx.log

finally i did found a solution. I just delete the .xmake directory in my appdata/locales directory, and i reinstalled the project with my Lua file, and xmake was able to install the package :) !

Sorry if i bothered you :) ! And thank you for your guidance anyway ;) 

![image](https://user-images.githubusercontent.com/91457036/183288974-99840fa1-7846-4995-8a43-2a785e79ac10.png)

**Reply by Kleb0** [reply]
![image](https://user-images.githubusercontent.com/91457036/183289005-af95b50a-a335-477e-ae06-5a3a25465f7b.png)
